packages/lib/file-api-driver-local.ts

```diff
@@ -13,11 +13,39 @@ const { basicDelta } = require('./file-api');
 // check that it is indeed the problem, check log-database.txt of both clients, search for the note ID, and most likely both notes
 // will have been modified at the same exact second at some point. If not, it's another bug that needs to be investigated.
 
-class FileApiDriverLocal {
-	fsErrorToJsError_(error, path = null) {
+// this is the format that fs-driver-rn would return
+type StdStatResult = {
+	birthtime: Date; // Created date
+	mtime: Date; // Last modified date
+	isDirectory: ()=> boolean; // Is the file a directory?
+	path: string;
+	size: number; // Size in bytes
+};
+
+type MetaData = {
+	path: string;
+	/** in milliseconds */
+	updated_time: number;
+	isDir: boolean;
+};
+
+export type FileOptions = {encoding?: string} & (
+	| {
+		source: 'file';
+		path: string;
+	}| {
+		target: 'file';
+		path: string;
+	}
+);
+
```

Comment:
I can't read that, sorry. Please either give it no type or make it clearer what it is

Reply:
```suggestion
export type FileOptions =
{
	encoding?: string;
	path: string;
} & ({ source: 'file' } | { target: 'file' });
```

Is this more clear?
What I did was just create a type from the usage already inside the file

Reply:
```suggestion
export type FileOptions =
{
	encoding?: string;
	path?: string;
	target?: 'file';
	source?: 'file';
};
```
well since I am restoring the original options, i suggest this